capabilities/index/index.go

```diff
@@ -0,0 +1,114 @@
+package index
+
+import (
+	_ "embed"
+	"fmt"
+
+	ipldprime "github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	ipldschema "github.com/ipld/go-ipld-prime/schema"
+	"github.com/storacha/go-libstoracha/capabilities/types"
+	"github.com/storacha/go-ucanto/core/ipld"
+	"github.com/storacha/go-ucanto/core/result/failure"
+	"github.com/storacha/go-ucanto/core/schema"
+	"github.com/storacha/go-ucanto/ucan"
+	"github.com/storacha/go-ucanto/validator"
+)
+
+//go:embed index.ipldsch
+var assertSchema []byte
+
+var assertTS = mustLoadTS()
+
+func mustLoadTS() *ipldschema.TypeSystem {
+	ts, err := ipldprime.LoadSchemaBytes(assertSchema)
+	if err != nil {
+		panic(fmt.Errorf("loading assert schema: %w", err))
+	}
+	return ts
+}
+
+func AddCaveatsType() ipldschema.Type {
+	return assertTS.TypeByName("AddCaveats")
+}
+
+var IndexAbility = "space/index/*"
```

Comment:
Humm, if this is for a space, then perhaps we should define a new dir in https://github.com/storacha/go-libstoracha/tree/main/capabilities/space for this type?